### PR TITLE
dreprecate distutils and move to setuptools

### PR DESCRIPTION
**Pull request by FantasqueX**
Hi there! I'm packaging mulpyplexer into an ArchLinux package and find a warning "DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives". I suggest deprecate distutils and move to setuptools to prevent it from not working in Python 3.12 :)

### CODE REVIEW

**Comment by twizmwazin** [comment]
If we are going to depend on setuptools, the correct way to do so would be to also add a PEP 518 pyproject.toml file.

**Comment by FantasqueX** [comment]
Yes, I think we should add a pyproject.toml file according to PEP 517 and PEP 518. However, I cannot find pyproject.toml in other projects of ANGR and I'm not  sure whether I should add that. What's more, according to the official [tutorial](https://packaging.python.org/en/latest/tutorials/packaging-projects/),  a static setup.cfg is more recommended than a dynamic one. I think I should do the minimal modification to keep future compatibility, however, if the maintainer thinks it's OK to move further, I will finish those things in this PR.

**Comment by twizmwazin** [comment]
@FantasqueX Just added [one](https://github.com/angr/archinfo/blob/master/pyproject.toml)! Defining parameters in a setup.cfg would be a further improvement, but not necessary here, that'd be best left to a separate PR since it wouldn't break compatibility, which this change without a pyproject.toml may.

**Comment by FantasqueX** [comment]
OK. I have added pyproject.toml. If you think it's ready to merge, I will squash those commits.

**Review by twizmwazin — APPROVED** [review]
I'll let @zardus hit merge since this is his project.

**Comment by FantasqueX** [comment]
Finshed squashing commits. Waiting for merging :)

**Comment by zardus** [comment]
Thanks!

**Pull request merged**
